src/core/tests/type_prop/binary_elementwise.cpp

```diff
@@ -445,3 +445,316 @@ TEST(type_prop, logic_arith_compare_partial_et) {
     ASSERT_EQ(test_logical_not(element::boolean)->get_element_type(), element::boolean);
     ASSERT_EQ(test_logical_not(element::dynamic)->get_element_type(), element::dynamic);
 }
+
+namespace {
+template <class T>
+shared_ptr<op::v1::Reshape> createReshapeSubgraph(PartialShape param_shape,
+                                                  shared_ptr<op::Constant> constant_op,
+                                                  bool const_rhs = true) {
+    auto param = make_shared<op::Parameter>(element::f32, param_shape);
+    auto shape_of = make_shared<op::v3::ShapeOf>(param);
+    auto cast_fp = make_shared<op::Convert>(shape_of, element::f32);
+
+    Output<Node> op;
+    if (const_rhs)
+        op = make_shared<T>(cast_fp, constant_op);
+    else
+        op = make_shared<T>(constant_op, cast_fp);
+
+    auto cast_int = make_shared<op::Convert>(op, element::i32);
+    auto r = make_shared<op::v1::Reshape>(param, cast_int, false);
+
+    return r;
+}
+}  // namespace
+
+TEST(type_prop, interval_value_propagation_add) {
+    PartialShape op_shape{Dimension(-1), Dimension(2, -1), Dimension(-1, 6), Dimension(7, 10), Dimension(7, 10), 5};
+    const auto const_op = op::Constant::create(element::f32, {6}, {2, 3, 4, 5, -5, 6});
+    {  // const rhs
+        const auto reshape = createReshapeSubgraph<op::v1::Add>(op_shape, const_op);
+        EXPECT_EQ(reshape->get_element_type(), element::f32);
+        EXPECT_EQ(
+            reshape->get_output_partial_shape(0),
+            PartialShape({-1, /* Dimension(5, -1) */ -1, Dimension(4, 10), Dimension(12, 15), Dimension(2, 5), 11}));
```

Comment:
In all of the operations, lower_bounds are not propagated, if upper_bounds are not set.
@jane-intel Is it correct?

Reply:
Could you describe the logic behind such propagation?

Reply:
I was under impression that we can calculate the lower_bound for Dimension with the same logic as for upper_bound. In this case for 
Add -> (2, -1) + (3, 3) = (5, -1) as Add -> (-1, 6) + (4, 4) = (4, 10)

Reply:
this is just because not all binary elementwise operations could have such logic.
this is the safest I could get to make universal bound evaluation.

If we calculate Multiply this wouldn't work.

(1, 10) x (-4, -1) = (-1, -40)
But if we would consider having only lower bounds 1 x -4 -- we would estimate output lower bound as -4 which is incorrect

Reply:
Ticket for the discussed bounds evaluation improvements: 95040